contrib/linux/actions/dig.py

```diff
@@ -39,11 +41,25 @@ def run(self, rand, count, nameserver, hostname, queryopts):
             cmd_args.append('+' + v)
 
         cmd_args.append(hostname)
-        result_list = filter(None, subprocess.Popen(cmd_args,
-                                                    stderr=subprocess.PIPE,
-                                                    stdout=subprocess.PIPE)
-                                             .communicate()[0]
-                                             .split('\n'))
+
+        try:
+            result_list = filter(None, subprocess.Popen(cmd_args,
+                                                        stderr=subprocess.PIPE,
+                                                        stdout=subprocess.PIPE)
+                                 .communicate()[0]
+                                 .split('\n'))
+
+        # NOTE: Python3 supports the FileNotFoundError, the errono.ENOENT is for py2 compat
+        # for Python3:
+        # except FileNotFoundError as e:
+
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                return False, "Can\'t find dig installed in our path (Usually /usr/bin/dig). " \
```

Comment:
For exceptions that are easily resolved, like this, I usually try to give some instructions for fixing it right in the error message. For this, I would probably add something like:

> Can't find dig installed in the path (usually /usr/bin/dig). If dig isn't installed, you can install it with 'sudo yum install bind-utils' or 'sudo apt install dnsutils'.

Reply:
@blag  @armab all set.